internal/testgen/gen_http_const.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/Antonboom/testifylint/internal/checkers"
+)
+
+type HTTPConstTestsGenerator struct{}
+
+func (HTTPConstTestsGenerator) Checker() checkers.Checker {
+	return checkers.NewHTTPConst()
+}
+
+func (g HTTPConstTestsGenerator) TemplateData() any {
+	checker := g.Checker().Name()
+
+	return struct {
+		CheckerName       CheckerName
+		InvalidAssertions []Assertion
+		ValidAssertions   []Assertion
+		IgnoredAssertions []Assertion
+	}{
+		CheckerName: CheckerName(checker),
+		InvalidAssertions: []Assertion{
+			{
+				Fn:            "HTTPStatusCode",
+				Argsf:         `httpOK, "get", "/index", nil, 200`,
+				ReportMsgf:    checker + ": use net/http constants instead of value",
+				ProposedArgsf: `httpOK, http.MethodGet, "/index", nil, http.StatusOK`,
+			}, {
+				Fn:            "HTTPStatusCode",
+				Argsf:         `httpOK, "Get", "/index", nil, 200`,
+				ReportMsgf:    checker + ": use net/http constants instead of value",
+				ProposedArgsf: `httpOK, http.MethodGet, "/index", nil, http.StatusOK`,
+			}, {
+				Fn:            "HTTPStatusCode",
+				Argsf:         `httpOK, "GET", "/index", nil, 200`,
+				ReportMsgf:    checker + ": use net/http constants instead of value",
+				ProposedArgsf: `httpOK, http.MethodGet, "/index", nil, http.StatusOK`,
+			}, {
+				Fn:            "HTTPBodyContains",
+				Argsf:         `httpHelloName, "GET",  "/", url.Values{"name": []string{"World"}}, "Hello, World!"`,
+				ReportMsgf:    checker + ": use net/http constants instead of value",
+				ProposedArgsf: `httpHelloName, http.MethodGet,  "/", url.Values{"name": []string{"World"}}, "Hello, World!"`,
+			},
+		},
+		ValidAssertions: []Assertion{
+			{Fn: "HTTPStatusCode", Argsf: `httpOK, http.MethodGet, "/index", nil, http.StatusOK`},
+			{Fn: "HTTPBodyContains", Argsf: `httpHelloName, http.MethodGet,  "/", url.Values{"name": []string{"World"}}, "Hello, World!"`},
+		},
+		IgnoredAssertions: []Assertion{},
```

Comment:
I would add to the ignored assertions an unofficial code and method to make sure code won't catch things it shouldn't

                                Argsf:         `httpOK, "FOO", "/index", nil, 123`,

Reply:
@mmorel-35 I don't know if you saw this one

Reply:
oh you are editing in a browser, I didn't get it.

I can take care of it

Reply:
I think you should add me to maintainer/users of your project. 

```console
$ git push --no-verify --force-with-lease mmorel 
ERROR: Permission to mmorel-35/testifylint.git denied to ccoVeille.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights and the repository exists.
```

Anton can do updates, but because he is a maintainer of Antonboom/testifylint, he didn't grant us to be maintainers of ` Antonboom/testifylint`.

Reply:
Same error message

Reply:
I had to click on the email I didn't notice.

It works now